fix(messenger_communities): block messages and reactions to token gated or spectated communities

shinnok opened the pull request:
Right now, if the QML UI control checks are disabled, the user can send messages to spectated, non-joined and, possibly, token-gated communities with _invalid tokens_?

Right now the PR is in draft stage with possible code culprits.

---
Closes https://github.com/status-im/status-desktop/issues/11915

ghost commented:
Hey @shinnok, and thank you so much for making your first pull request in status-go! :heart: Please help us make your experience better by filling out this brief questionnaire https://goo.gl/forms/uWqNcVpVz7OIopXg2

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (96)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | bfb1846a | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/1/) | 2023-09-26 15:36:49 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230926-153250-bfb184-pr4064.zip) |
| :heavy_check_mark: | bfb1846a | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/1/) | 2023-09-26 15:37:24 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230926-153250-bfb184-pr4064.zip) |
| :heavy_check_mark: | bfb1846a | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/1/) | 2023-09-26 15:38:55 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230926-153250-bfb184-pr4064.aar) |
| :heavy_multiplication_x: | bfb1846a | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/1/) | 2023-09-26 15:51:06 | ~18 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 454911cd | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/2/) | 2023-10-04 15:11:51 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231004-150829-454911-pr4064.zip) |
| :heavy_check_mark: | 454911cd | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/2/) | 2023-10-04 15:12:14 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231004-150829-454911-pr4064.zip) |
| :heavy_check_mark: | 454911cd | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/2/) | 2023-10-04 15:13:21 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231004-150824-454911-pr4064.aar) |
| :heavy_multiplication_x: | 454911cd | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/2/) | 2023-10-04 15:28:23 | ~19 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | d7134bac | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/3/) | 2023-10-05 10:44:03 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231005-104242-d7134b-pr4064.zip) |
| :heavy_check_mark: | d7134bac | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/3/) | 2023-10-05 10:44:30 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231005-104239-d7134b-pr4064.aar) |
| :heavy_check_mark: | d7134bac | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/3/) | 2023-10-05 10:46:34 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231005-104239-d7134b-pr4064.zip) |
| :heavy_multiplication_x: | d7134bac | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/3/) | 2023-10-05 10:58:28 | ~15 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 9d05479c | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/4/) | 2023-10-05 15:06:28 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231005-150430-9d0547-pr4064.aar) |
| :heavy_check_mark: | 9d05479c | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/4/) | 2023-10-05 15:07:25 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231005-150434-9d0547-pr4064.zip) |
| :heavy_check_mark: | 9d05479c | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/4/) | 2023-10-05 15:07:51 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231005-150434-9d0547-pr4064.zip) |
| :heavy_multiplication_x: | 9d05479c | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/4/) | 2023-10-05 15:23:15 | ~18 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 8e179c57 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/5/) | 2023-10-06 07:47:42 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231006-074613-8e179c-pr4064.zip) |
| :heavy_check_mark: | 8e179c57 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/5/) | 2023-10-06 07:48:07 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231006-074608-8e179c-pr4064.aar) |
| :heavy_check_mark: | 8e179c57 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/5/) | 2023-10-06 07:48:58 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231006-074608-8e179c-pr4064.zip) |
| :heavy_multiplication_x: | 8e179c57 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/5/) | 2023-10-06 08:01:38 | ~15 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/5//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 22a19845 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/6/) | 2023-10-06 14:31:14 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/6//consoleText) |
| :heavy_check_mark: | 22a19845 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/6/) | 2023-10-06 14:31:28 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231006-143010-22a198-pr4064.zip) |
| :heavy_check_mark: | 22a19845 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/6/) | 2023-10-06 14:31:38 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231006-143005-22a198-pr4064.aar) |
| :heavy_check_mark: | 22a19845 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/6/) | 2023-10-06 14:32:58 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231006-143005-22a198-pr4064.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 39789d9a | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/7/) | 2023-10-09 09:37:05 | ~59 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/7//consoleText) |
| :heavy_check_mark: | 39789d9a | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/7/) | 2023-10-09 09:37:46 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231009-093605-39789d-pr4064.aar) |
| :heavy_check_mark: | 39789d9a | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/7/) | 2023-10-09 09:37:51 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231009-093605-39789d-pr4064.zip) |
| :heavy_check_mark: | 39789d9a | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/7/) | 2023-10-09 09:39:13 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231009-093605-39789d-pr4064.zip) |
| | | | | | | |
| :heavy_check_mark: | dace1b28 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/8/) | 2023-10-11 11:08:59 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231011-110723-dace1b-pr4064.zip) |
| :heavy_multiplication_x: | dace1b28 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/8/) | 2023-10-11 11:09:41 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/8//consoleText) |
| :heavy_check_mark: | dace1b28 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/8/) | 2023-10-11 11:10:56 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231011-110723-dace1b-pr4064.zip) |
| :heavy_check_mark: | dace1b28 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/8/) | 2023-10-11 11:13:00 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231011-110723-dace1b-pr4064.aar) |
| | | | | | | |
| :heavy_check_mark: | 372bc118 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/9/) | 2023-10-11 13:41:51 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231011-133943-372bc1-pr4064.zip) |
| :heavy_check_mark: | 372bc118 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/9/) | 2023-10-11 13:42:15 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231011-133939-372bc1-pr4064.aar) |
| :heavy_check_mark: | 372bc118 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/9/) | 2023-10-11 13:43:24 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231011-133943-372bc1-pr4064.zip) |
| :heavy_multiplication_x: | 372bc118 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/9/) | 2023-10-11 13:55:51 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/9//consoleText) |
| | | | | | | |
| :heavy_check_mark: | cfd7a3e5 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/10/) | 2023-10-12 09:14:59 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231012-091329-cfd7a3-pr4064.zip) |
| :heavy_check_mark: | cfd7a3e5 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/10/) | 2023-10-12 09:15:31 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231012-091329-cfd7a3-pr4064.aar) |
| :heavy_check_mark: | cfd7a3e5 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/10/) | 2023-10-12 09:16:39 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231012-091329-cfd7a3-pr4064.zip) |
| :heavy_multiplication_x: | cfd7a3e5 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/10/) | 2023-10-12 09:28:06 | ~14 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/10//consoleText) |
| | | | | | | |
| :heavy_check_mark: | fd675b8f | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/11/) | 2023-10-12 14:32:12 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231012-143040-fd675b-pr4064.aar) |
| :heavy_check_mark: | fd675b8f | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/11/) | 2023-10-12 14:33:37 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231012-143044-fd675b-pr4064.zip) |
| :heavy_check_mark: | fd675b8f | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/11/) | 2023-10-12 14:33:59 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231012-143044-fd675b-pr4064.zip) |
| :heavy_multiplication_x: | fd675b8f | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/11/) | 2023-10-12 14:45:40 | ~14 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/11//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 67badfca | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/12/) | 2023-10-12 15:51:23 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231012-154945-67badf-pr4064.zip) |
| :heavy_check_mark: | 67badfca | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/12/) | 2023-10-12 15:51:48 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231012-154941-67badf-pr4064.aar) |
| :heavy_check_mark: | 67badfca | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/12/) | 2023-10-12 15:52:41 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231012-154941-67badf-pr4064.zip) |
| :heavy_multiplication_x: | 67badfca | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/12/) | 2023-10-12 16:06:11 | ~16 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/12//consoleText) |
| | | | | | | |
| :heavy_check_mark: | f388ab28 | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/13/) | 2023-10-13 08:11:00 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231013-080939-f388ab-pr4064.zip) |
| :heavy_check_mark: | f388ab28 | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/13/) | 2023-10-13 08:11:13 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231013-080934-f388ab-pr4064.aar) |
| :heavy_check_mark: | f388ab28 | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/13/) | 2023-10-13 08:12:58 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231013-080939-f388ab-pr4064.zip) |
| :heavy_multiplication_x: | f388ab28 | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/13/) | 2023-10-13 08:24:31 | ~14 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/13//consoleText) |
| :heavy_multiplication_x: | f388ab28 | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/14/) | 2023-10-13 09:55:49 | ~9 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/14//consoleText) |
| :heavy_multiplication_x: | f388ab28 | [#15](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/15/) | 2023-10-13 10:27:56 | ~13 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/15//consoleText) |
| :heavy_multiplication_x: | f388ab28 | [#16](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/16/) | 2023-10-13 11:07:58 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/16//consoleText) |
| :heavy_multiplication_x: | f388ab28 | [#17](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/17/) | 2023-10-13 14:23:25 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/17//consoleText) |
| :heavy_multiplication_x: | f388ab28 | [#18](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/18/) | 2023-10-16 09:56:56 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/18//consoleText) |
| :heavy_multiplication_x: | f388ab28 | [#19](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/19/) | 2023-10-17 06:57:41 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/19//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 3d5ab9c4 | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/14/) | 2023-10-17 09:03:43 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231017-090214-3d5ab9-pr4064.zip) |
| :heavy_check_mark: | 3d5ab9c4 | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/14/) | 2023-10-17 09:03:56 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231017-090209-3d5ab9-pr4064.aar) |
| :heavy_multiplication_x: | 3d5ab9c4 | [#20](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/20/) | 2023-10-17 09:04:18 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/20//consoleText) |
| :heavy_check_mark: | 3d5ab9c4 | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/14/) | 2023-10-17 09:05:34 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231017-090214-3d5ab9-pr4064.zip) |
| | | | | | | |
| :heavy_check_mark: | 0453246b | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/15/) | 2023-10-17 13:40:48 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231017-133935-045324-pr4064.zip) |
| :heavy_check_mark: | 0453246b | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/15/) | 2023-10-17 13:40:59 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231017-133931-045324-pr4064.aar) |
| :heavy_check_mark: | 0453246b | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/15/) | 2023-10-17 13:43:18 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231017-133931-045324-pr4064.zip) |
| :heavy_multiplication_x: | 0453246b | [#21](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/21/) | 2023-10-17 14:12:12 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/21//consoleText) |
| :heavy_multiplication_x: | 0453246b | [#22](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/22/) | 2023-10-17 15:05:57 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/22//consoleText) |
| :heavy_multiplication_x: | 0453246b | [#23](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/23/) | 2023-10-18 13:28:53 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/23//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 794607f9 | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/16/) | 2023-10-18 15:48:06 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231018-154644-794607-pr4064.zip) |
| :heavy_check_mark: | 794607f9 | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/16/) | 2023-10-18 15:49:59 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231018-154644-794607-pr4064.zip) |
| :heavy_check_mark: | 794607f9 | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/16/) | 2023-10-18 15:52:13 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231018-154644-794607-pr4064.aar) |
| :heavy_multiplication_x: | 794607f9 | [#24](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/24/) | 2023-10-18 16:12:30 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/24//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 48ddf615 | [#25](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/25/) | 2023-10-20 14:17:00 | ~54 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/25//consoleText) |
| :heavy_check_mark: | 48ddf615 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/17/) | 2023-10-20 14:17:27 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231020-141605-48ddf6-pr4064.zip) |
| :heavy_check_mark: | 48ddf615 | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/17/) | 2023-10-20 14:17:49 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231020-141600-48ddf6-pr4064.aar) |
| :heavy_check_mark: | 48ddf615 | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/17/) | 2023-10-20 14:19:28 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231020-141600-48ddf6-pr4064.zip) |
| | | | | | | |
| :heavy_check_mark: | 5eb92343 | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/18/) | 2023-10-20 14:41:43 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231020-144030-5eb923-pr4064.zip) |
| :heavy_check_mark: | 5eb92343 | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/18/) | 2023-10-20 14:42:16 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231020-144030-5eb923-pr4064.aar) |
| :heavy_check_mark: | 5eb92343 | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/18/) | 2023-10-20 14:43:44 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231020-144030-5eb923-pr4064.zip) |
| :heavy_multiplication_x: | 5eb92343 | [#26](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/26/) | 2023-10-20 14:49:52 | ~9 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/26//consoleText) |
| :heavy_multiplication_x: | 5eb92343 | [#27](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/27/) | 2023-10-20 15:33:49 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/27//consoleText) |
| :heavy_check_mark: | 5eb92343 | [#28](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/28/) | 2023-10-23 08:29:07 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/28//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 5f30482d | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/19/) | 2023-10-23 08:47:05 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231023-084411-5f3048-pr4064.zip) |
| :heavy_check_mark: | 5f30482d | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/19/) | 2023-10-23 08:47:50 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231023-084411-5f3048-pr4064.zip) |
| :heavy_check_mark: | 5f30482d | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/19/) | 2023-10-23 08:48:41 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231023-084411-5f3048-pr4064.aar) |
| :heavy_check_mark: | 5f30482d | [#29](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/29/) | 2023-10-23 09:16:30 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/29//consoleText) |
| | | | | | | |
| :heavy_check_mark: | b69d5b07 | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/20/) | 2023-10-23 09:19:36 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231023-091756-b69d5b-pr4064.zip) |
| :heavy_check_mark: | b69d5b07 | [#20](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/20/) | 2023-10-23 09:20:15 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231023-091756-b69d5b-pr4064.aar) |
| :heavy_check_mark: | b69d5b07 | [#20](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/20/) | 2023-10-23 09:21:34 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231023-091756-b69d5b-pr4064.zip) |
| :heavy_multiplication_x: | b69d5b07 | [#30](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/30/) | 2023-10-23 09:50:41 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/30//consoleText) |
| :heavy_check_mark: | b69d5b07 | [#31](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/31/) | 2023-10-23 10:45:38 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/31//consoleText) |
| | | | | | | |
| :heavy_check_mark: | a69e1cbf | [#21](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/21/) | 2023-10-24 14:06:02 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231024-140424-a69e1c-pr4064.zip) |
| :heavy_check_mark: | a69e1cbf | [#21](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/21/) | 2023-10-24 14:06:39 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231024-140419-a69e1c-pr4064.aar) |
| :heavy_multiplication_x: | a69e1cbf | [#32](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/32/) | 2023-10-24 14:07:16 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/32//consoleText) |
| :heavy_check_mark: | a69e1cbf | [#21](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/21/) | 2023-10-24 14:07:52 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231024-140419-a69e1c-pr4064.zip) |
| :heavy_multiplication_x: | a69e1cbf | [#33](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/33/) | 2023-10-24 15:01:21 | ~27 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/33//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 55ba45e6 | [#22](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/22/) | 2023-10-24 15:54:53 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231024-155120-55ba45-pr4064.zip) |
| :heavy_check_mark: | 55ba45e6 | [#22](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/22/) | 2023-10-24 15:55:33 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231024-155120-55ba45-pr4064.aar) |
| :heavy_check_mark: | 55ba45e6 | [#22](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/22/) | 2023-10-24 15:58:05 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231024-155124-55ba45-pr4064.zip) |
| :heavy_multiplication_x: | 55ba45e6 | [#34](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/34/) | 2023-10-24 15:58:59 | ~7 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/34//consoleText) |
| :heavy_check_mark: | 55ba45e6 | [#35](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/35/) | 2023-10-24 16:43:25 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/35//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 4fe5a2c5 | [#23](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4064/23/) | 2023-10-25 10:35:41 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231025-103231-4fe5a2-pr4064.zip) |
| :heavy_check_mark: | 4fe5a2c5 | [#23](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4064/23/) | 2023-10-25 10:36:36 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231025-103231-4fe5a2-pr4064.zip) |
| :heavy_check_mark: | 4fe5a2c5 | [#23](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4064/23/) | 2023-10-25 10:37:02 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231025-103231-4fe5a2-pr4064.aar) |
| :heavy_check_mark: | 4fe5a2c5 | [#36](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/36/) | 2023-10-25 11:07:41 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4064/36//consoleText) |

jrainville commented on protocol/communities/community.go:
```diff
@@ -902,6 +902,7 @@ func (o *Community) Leave() {
 
 func (o *Community) Spectate() {
 	o.config.Spectated = true
+	o.config.Joined = false // this doesn't seem to fix anything, but looks proper to do
```
This is probably not needed. It defaults to false. We can keep it or not.

shinnok replied:
Reverted.

jrainville commented on protocol/communities/community.go:
```diff
@@ -1605,6 +1606,11 @@ func (o *Community) CanPost(pk *ecdsa.PublicKey, chatID string, grantBytes []byt
 	}
 
+	// this resolves the issue, however, I'm not sure how it should play with grants and tokens below
+	if !o.isMember() {
```
That looks right. We don't need to check the tokens or whatever if this condition doesn't pass. If someone is not a member, they cannot post.

jrainville commented on protocol/messenger.go:
```diff
@@ -2218,13 +2218,18 @@ func (m *Messenger) sendChatMessage(ctx context.Context, message *common.Message
+	}
 
-	} else if len(message.CommunityID) != 0 {
+	if len(message.CommunityID) != 0 {
```
I think this code path is only to send an invitation to the community. So it's probably not the culprit

shinnok replied:
I tried to split the logic here a bit, so it's more obvious what we're checking for. Will keep this bit.

jrainville reviewed (commented):
The line in `CanPost` seems like the right culprit for the Sender side.

We should also do a similar check on the receiver side.
The entry point for that is in `messenger_handlers.go` > `dispatchToHandler`. 
That's where we dispatch the received message to the right handler. 
In this case, it would be `handleChatMessageProtobuf`.

Pro tip: using `fmt.Println("...")`  is a good way to add logs to status-go that will show up in the console when running Status Desktop.

But even better is using the Go tests. You'll find a lot of useful tests in `community_test.go`. For example `TestCanPost`. `TestValidateRequestToJoin` might be a good one to validate that messages only work for those joined, but maybe it's better to create a new test case.

shinnok commented:
I can't get the `SendPinMessage()` call to go through even though I have `PinMessageAllMembersEnabled` set to **true**.

[`39789d9` (#4064)](https://github.com/status-im/status-go/pull/4064/commits/39789d9a80d212ccd611835a20ece38e61da1704#diff-d4f4b1d3dcfd1f9dda7f6d7a397650c30b499042a7770a1df9a6fe80d61f04feR441)

https://github.com/status-im/status-go/blob/fix/11915/send-message-to-gated-community/protocol/messenger_pin_messages.go#L42

Not sure how critical it is to get the `TestPinMessageInCommunityChat` working for this particular issue, thus I'm enabling this PR for review and also welcome to help on investigating why isn't the community message pin working in the test suite, when it is fine otherwise.

osmaczko commented on protocol/communities_messenger_test.go:
```diff
@@ -436,12 +436,37 @@ func (s *MessengerCommunitiesSuite) TestCommunityContactCodeAdvertisement() {
+	community.AdminSettings().PinMessageAllMembersEnabled = true // FIXME: this doesn't work
 
 	s.advertiseCommunityTo(community, s.alice)
+	s.alice.SpectateCommunity(community.ID()) // TODO: also test in stages like TestPostToCommunityChat
```
This line seems unnecessary, there is no strict requirement to spectate before joining.

osmaczko commented on protocol/communities_messenger_test.go:
```diff
@@ -436,12 +436,37 @@ func (s *MessengerCommunitiesSuite) TestCommunityContactCodeAdvertisement() {
 }
 
-func (s *MessengerCommunitiesSuite) TestPostToCommunityChat() {
+func (s *MessengerCommunitiesSuite) TestPinMessageInCommunityChat() {
```
No one can prevent a peer from sending a pinned message in the community channel (someone might modify the status-go source code for their client). We should also check on the receiver's side to ensure that pinned messages from non-approved members are rejected.

shinnok replied:
This is fixed for both sender and receiver in my testing. Check out the consolidated fixes in final commit dace1b28d8510bc9514ff4661928d2fb5c0302e6.

shinnok replied:
To expand on the matter a bit, both a community and individual chats of access permissions of these type:
https://github.com/status-im/status-go/blob/fix/11915/send-message-to-gated-community/protocol/protobuf/communities.pb.go

On top of that, there is the distinction that a user is a member, or not, of a community and then an individual chat. A user becomes a member of a Community when it is first advertised by another user B and A requests to join. Then what happens to A's request to join depends on _Community.Access.Permission_. After joining, when A attempt a post, the protocol code decides whether to allow a post, a reaction or a pin, based on _Community.Chat.Access.Permission._ Please correct if any of these is incorrect.

Now, where do grants fit in? As I could not find either documentation nor help pages describing what it is, nor the code hints seems to be implemented fully:

[`156bf39` (#4064)](https://github.com/status-im/status-go/pull/4064/commits/156bf3942a7d2ecf855628fa6aad82c03072f50d)

osmaczko commented on protocol/communities_messenger_test.go:
```diff
@@ -436,12 +436,37 @@ func (s *MessengerCommunitiesSuite) TestCommunityContactCodeAdvertisement() {
-func (s *MessengerCommunitiesSuite) TestPostToCommunityChat() {
+func (s *MessengerCommunitiesSuite) TestPinMessageInCommunityChat() {
 	community, chat := s.createCommunity()
+	community.AdminSettings().PinMessageAllMembersEnabled = true // FIXME: this doesn't work
```
This doesn't work because the changes aren't saved. When a community is advertised to the user, only the community ID is propagated. Then, when constructing the community message, it is fetched from the database.

shinnok replied:
Moved the `TestPinMessageInCommunityChat` and afferent investigation to https://github.com/status-im/status-go/pull/4139.

osmaczko reviewed (commented):
Looks good in general :+1: some minor things to address.

jrainville commented on protocol/messenger.go:
```diff
@@ -2241,11 +2244,6 @@ func (m *Messenger) sendChatMessage(ctx context.Context, message *common.Message
-		err := message.LoadAudio()
-		if err != nil {
-			return nil, err
-		}
```
The changes above can be reverted

shinnok replied:
Done. Check consolidated changes in dace1b28d8510bc9514ff4661928d2fb5c0302e6.

jrainville reviewed (commented):
Code looks good, though there is a small cleanup needed.

As we discussed in our 1-1, the Pin message test can be added in another PR, because it seems like the Setting to let users pin in a community is not being saved correctly on the receiver side, so it's worth another issue.

osmaczko commented:
> it seems like the Setting to let users pin in a community is not being saved correctly on the receiver side

I believe the root cause is different: https://github.com/status-im/status-go/pull/4064#discussion_r1350782701

jrainville commented:
> > it seems like the Setting to let users pin in a community is not being saved correctly on the receiver side
> 
> I believe the root cause is different: [#4064 (comment)](https://github.com/status-im/status-go/pull/4064#discussion_r1350782701)

Yeah I know. When I was in the 1-1 with him, we did add the `saveCommunity` call. Even though I checked that the owner's DB did have the right value afterwards, Alice's DB didn't have it. So, it seems like maybe members don,t save settings correctly?

Or maybe we do not publish community settings with `advertiseCommunity`?

osmaczko commented:
> > > it seems like the Setting to let users pin in a community is not being saved correctly on the receiver side
> > 
> > 
> > I believe the root cause is different: [#4064 (comment)](https://github.com/status-im/status-go/pull/4064#discussion_r1350782701)
> 
> Yeah I know. When I was in the 1-1 with him, we did add the `saveCommunity` call. Even though I checked that the owner's DB did have the right value afterwards, Alice's DB didn't have it. So, it seems like maybe members don,t save settings correctly?
> 
> Or maybe we do not publish community settings with `advertiseCommunity`?

The `saveCommunity` should be executed on the control node side before it is advertised to the users. The admin settings are a part of the `CommunityDescription`. The description sent by the control node should be identical on the receiver side because the proto lib handles unmarshaling. I would try to investigate what admin settings (`CommunityDescription`) is distributed exactly.

shinnok commented:
I've created issue https://github.com/status-im/status-go/issues/4138 to further investigate the channel message pin issue and forked the new test method `TestPinMessageInCommunityChat` to PR https://github.com/status-im/status-go/pull/4139.

---

The current squashed commit contains the sufficient changes to block messages, reactions and pins to channels for which the user is not a member, both for him and other members. Please review again.

jrainville commented on protocol/communities/community_test.go:
```diff
@@ -483,18 +483,30 @@ func (s *CommunitySuite) TestValidateRequestToJoin() {
+// | NO-MEMBERSHIP  | x         | Yes         | -       | Yes      |
+// | NO-MEMBERSHIP  | x         | No          | -       | No       |
+// | NO-MEMBERSHIP  | x         | No          | Valid   | Yes?     |
+// | NO-MEMBERSHIP  | x         | No          | Invalid | No       |
```
@cammellos can you shortly explain what a grant is? I feel like it's probably something from an old requirement, but I don't remember seeing it in any of the new requirements or code.

I'm of the opinion that if someone is not part of the member list, they shouldn't be able to post ever. However, this test below seems to imply that with a grant, it should be possible. Is that still something we want? If not, might as well remove all the grant code I guess?

cc @John-44 to clarify requirements. Basically, am I correct that you 100% need to be a member of a community/chat to be able to post to it?

jrainville replied:
I just discussed this with @cammellos and he said:
> so, the idea is that when you join a community, you get a "grant"
and that's a proof you have been added
so when you send a message, you send also the grant
so other clients can optimistically process the message, if the community description hasn't reached them yet

But it turns out we never fully supported it anyway:
> we never really supported it

He then told me it's fine to remove it.

I'll open an issue in a few.

@shinnok that means you can just set the grant tests to `canPost: false`, so that the test will pass. Then someone will have the chore of removing the grant code that's no longer useful later.

jrainville replied:
I added the issue to remove the grant code here: https://github.com/status-im/status-go/issues/4180

shinnok replied:
I adjusted the expected result for the sub-tests that are supposed to fail in TestCanPost().

jrainville commented on protocol/communities/community.go:
```diff
@@ -1742,6 +1742,12 @@ func (o *Community) CanPost(pk *ecdsa.PublicKey, chatID string, grantBytes []byt
 	}
 
+	// if not a member can't post or react on message
+	if !o.IsMemberInChat(pk, chatID) {
```
The condition `if o.config.CommunityDescription.Members == nil {` at line 1780 (the comment on it seems not ok) should be moved above this one or merged with this one.

jrainville reviewed (commented):
Looks good, the order of conditions in `canPost` can be improved a little

osmaczko commented on protocol/messenger_activity_center_test.go:
Thanks for cleanup.

osmaczko reviewed (approved):
:+1:

shinnok commented:
Attaching a video showing the changes in effect, the top left Status instance has adjustments in this PR in place, the bottom right doesn't, but instead has the QML UI checks disabled for sending, reacting and pinning. This effectively verifies that the receiving side effectively blocks non-member actions, however, once applied, this patch also denies non-member actions on the sender's side for efficiency. Testing this exact same workflow without this patch on either side, but with the [UI checks disabled](https://github.com//status-im/status-desktop/tree/fix/11915/send-message-to-gated-community), will allow a non-member to send, react and pin before being approved in a on-request community or before spectating for no-request community.

https://github.com/status-im/status-go/assets/544446/870f70ec-0a10-4ba0-97a5-327b3f95243d

* video is encoded to H.265/HEVC due to size/length, might help to download and play with VLC/Mplayer instead of the browser *

shinnok committed:
Check community>chat membership according to new requirement

Which specifies that if a user is not a community member & a
chat member, he can't post, react or pin messages in that chat.

Notes:
- also fix&cleanup associated failing tests.
- refactor Community.CanPost() to reflect the new requirement.
- grant code is not fully implemented and is to be removed later.

Fixes #11915
